File: src/main/java/be/breina/openrgb/model/Zone.java
package be.breina.openrgb.model;

import java.io.IOException;
import lombok.ToString;
import be.breina.openrgb.util.LittleEndianInputStream;

@ToString
public class Zone {

    private final String name;
    private final ZoneType type;
    private final int ledsMin, ledsMax, ledCount;
    private MatrixMap matrix;

    Zone(LittleEndianInputStream input) throws IOException {
        name = input.readAscii();
        type = ZoneType.values()[input.readInt()];
        ledsMin = input.readInt();
        ledsMax = input.readInt();
        ledCount = input.readInt();

        if (input.readUnsignedShort() > 0) {
            matrix = new MatrixMap(input);
        }
    }

    public String getName() {
        return name;
    }

    public ZoneType getType() {
        return type;
    }

    public int getLedsMin() {
        return ledsMin;
    }

    public int getLedsMax() {
        return ledsMax;
    }

    public int getLedCount() {
        return ledCount;
    }

    public MatrixMap getMatrix() {
        return matrix;
    }
}

File: src/main/java/be/breina/openrgb/model/Zone.kt
package be.breina.openrgb.model

import be.breina.openrgb.model.enumeration.ZoneType
import be.breina.openrgb.util.LittleEndianInputStream
import lombok.ToString

@ToString
class Zone internal constructor(
    val name: String?,
    val type: ZoneType,
    val ledsMin: Int,
    val ledsMax: Int,
    val ledCount: Int,
    var matrix: MatrixMap? = null,
) {

    companion object {
        fun from(input: LittleEndianInputStream): Zone {
            val name = input.readAscii()
            val type = ZoneType.values()[input.readInt()]
            val ledsMin = input.readInt()
            val ledsMax = input.readInt()
            val ledCount = input.readInt()
            val matrix: MatrixMap? = MatrixMap.from(input)

            return Zone(name, type, ledsMin, ledsMax, ledCount, matrix)
        }
    }
}